### PR TITLE
at86rf2xx: Add support for channel page

### PR DESCRIPTION
**Pull request by DipSwitch**
Followup PR for @gebart 's page and channel selection (#4191) since he's traveling and it would be nice to see this in the next release. :)

Also includes #4411 to making Travis happy with this PR.

### CODE REVIEW

**Comment by DipSwitch** [comment]
If merged closes: #3452 and #4191.

I can't set labels or assign the PR to people, so someone else needs to copy the labels of the gebarts PR. :)

For completeness the only interaction I had was:

	git checkout gebart/pr/at86rf2xx-phy-mode
	git checkout -b pr/at86rf2xx-phy-mode
	git fetch upstream
	git merge upstream/master
	git commit
	git merge fix_doxygen_at86rf2xx
	git rebase upstream/master
	git rebase -i HEAD~6
	git push --set-upstream origin pr/at86rf2xx-phy-mode

So I guess it's still ACk'ed.

File: drivers/at86rf2xx/at86rf2xx_netdev.c [hunk 1]
```diff
@@ -442,6 +442,14 @@ static int _get(gnrc_netdev_t *device, netopt_t opt, void *val, size_t max_len)
             ((uint8_t *)val)[0] = at86rf2xx_get_chan(dev);
             return sizeof(uint16_t);
 
+        case NETOPT_CHANNEL_PAGE:
+            if (max_len < sizeof(uint16_t)) {
```
Comment (thomaseichinger) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: I'm sorry if I'm missing something here, but why do we expect a `uint16_t` pointer here? Especially as we're only using the half of it.

**Comment by thomaseichinger** [comment]
@DipSwitch I'm ok with this PR to be merged for this release. But as mentioned in #4419 this PR also increases code size for e.g. rf23* devices by ~100bytes. Could you give #4419 a try and check I kept logic intact?

**Comment by DipSwitch** [comment]
I will try it tomorrow morning. But I can only test it for the 212(B). I don't have the other transceivers at my disposal.

**Comment by thomaseichinger** [comment]
Great! Will give this one  another try tomorrow morning too But I think we're ready to go then.

**Comment by cgundogan** [comment]
tested with two mulle boards. I was able to set the page and verify that pinging does work if both have the same page and that ping fails if both have different pages. ACK

**Comment by thomaseichinger** [comment]
@cgundogan Thank you! ACK & go.

**Comment by jfischer-phytec-iot** [comment]
I suspect this PR has made the radio support for samr21-xpro broken. 
Can anyone confirm?

**Comment by DipSwitch** [comment]
I tested it for more than a month or so, but my MCU (STM32 with AT86RF212) is running 36MHz so maybe a race condition somewhere although that doesn't make sence I guess. What transceiver do you use?

File: drivers/at86rf2xx/at86rf2xx_internal.c [hunk 2]
```diff
@@ -135,3 +135,78 @@ void at86rf2xx_hardware_reset(at86rf2xx_t *dev)
...
+
+    uint8_t trx_ctrl2 = at86rf2xx_reg_read(dev, AT86RF2XX_REG__TRX_CTRL_2);
+    uint8_t rf_ctrl0 = at86rf2xx_reg_read(dev, AT86RF2XX_REG__RF_CTRL_0);
+
+    /* Clear previous configuration for PHY mode */
+    trx_ctrl2 &= ~(AT86RF2XX_TRX_CTRL_2_MASK__FREQ_MODE);
+    /* Clear previous configuration for GC_TX_OFFS */
+    rf_ctrl0 &= ~AT86RF2XX_RF_CTRL_0_MASK__GC_TX_OFFS;
+    /* Clear previous configuration for channel number */
+    phy_cc_cca &= ~(AT86RF2XX_PHY_CC_CCA_MASK__CHANNEL);
```
Comment (thomaseichinger) [inline drivers/at86rf2xx/at86rf2xx_internal.c]: .

**Comment by thomaseichinger** [comment]
Sorry, I forgot to comment after testing in the afternoon. Yes, rf233 and rf231 are broken in current master. From looking at the code, I suspect the line [commented](https://github.com/RIOT-OS/RIOT/pull/4412/files#r47161357) should be moved outside the rf212b guard.

**Comment by OlegHahm** [comment]
broken on master and 2015.12-branch I assume?

**Comment by DipSwitch** [comment]
I think it's in yes.

**Comment by OlegHahm** [comment]
Is there already a fix for that PR'ed somewhere?

**Comment by OlegHahm** [comment]
see https://github.com/RIOT-OS/RIOT/pull/4482 and https://github.com/RIOT-OS/RIOT/pull/4484

**Pull request merged**